localgov_base.theme

```diff
@@ -65,6 +79,46 @@ function localgov_base_preprocess_page(&$variables) {
   if (!isset($variables['localgov_base_remove_css'])) {
     $variables['#attached']['library'][] = 'localgov_base/global';
   }
+
+  // We have a theme setting boolean for removing the pink background from
+  // unpublished content. If that setting is false,
+  // we will add the 'unpublished' library to our page.
+  dpm(theme_get_setting('localgov_base_add_unpublished_background_colour'));
+  if (theme_get_setting('localgov_base_add_unpublished_background_colour') == TRUE) {
+    $variables['#attached']['library'][] = 'localgov_base/unpublished-bg';
+  }
+}
+
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function localgov_base_preprocess_node(&$variables) {
+  $node_status = $variables['node']->isPublished();
+  if ($node_status === FALSE) {
+    if (theme_get_setting('localgov_base_add_draft_note_to_unpublished_content') == TRUE) {
+      $variables['label'] = t('[Draft]') . " " . $variables['node']->label();
+    }
+  }
+}
+
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function localgov_base_preprocess_block(&$variables) {
+  $route_match = Drupal::routeMatch();
+  // Check if we are on a node page.
+  if ($route_match->getRouteName() == 'entity.node.canonical') {
+    $node = $route_match->getParameter('node');
+    $node_status = $node->isPublished();
+    if ($node_status === FALSE) {
+      if (theme_get_setting('localgov_base_add_draft_note_to_unpublished_content') == TRUE) {
+        if ($variables['plugin_id'] === 'localgov_page_header_block') {
```

Comment:
I'd check this first, as it's the most specific condition. It means that in most cases, we'll exit earlier and spend less time checking all the other conditions when we're not looking at this block.